### PR TITLE
fix getTrip, add delete method, add userId field to trip

### LINKED ISSUE #127: Implement delete trip method in dart

### LINKED ISSUE #137: fix getTrip error
fix error when trying to get trip through REST api

### PR DESCRIPTION
**Pull request by cli1903**

### CODE REVIEW

**Review by chuckluv — APPROVED** [review]
LGTM

File: frontend/src/main/java/com/google/tripmeout/frontend/servlet/TripServlet.java [hunk 1]
```diff
@@ -17,7 +17,7 @@
   private static final FluentLogger logger = FluentLogger.forEnclosingClass();
 
   private static final long serialVersionUID = 1L;
-  private static final String APPLICATION_JSON_CONTENT_TYPE = "application/json;";
+  private static final String APPLICATION_JSON_CONTENT_TYPE = "application/json";
```
Comment (anfee1) [inline frontend/src/main/java/com/google/tripmeout/frontend/servlet/TripServlet.java:20]: I am going to assume this was causing a problem, do you know why?

**Review by anfee1 — APPROVED** [review]
LGTM

File: frontend/src/main/java/com/google/tripmeout/frontend/servlet/TripServlet.java (hunk 1 above)
Comment (cli1903) [reply frontend/src/main/java/com/google/tripmeout/frontend/servlet/TripServlet.java:20]: based on what I read online, it seems like the flutter html package just doesn't like trailing semi-colons

Comment (anfee1) [reply frontend/src/main/java/com/google/tripmeout/frontend/servlet/TripServlet.java:20]: ok

**Review by KoopaKing — APPROVED** [review]
Approving this, but this should be separate PRs. These changes are all independent.

**Pull request merged**